### PR TITLE
JENKINS-61207 System read - logs

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-61207](https://issues.jenkins-ci.org/browse/JENKINS-61207).

Adds system read permission support to log recorder manager.

### Screenshots

<details>

<summary>All log messages</summary>

image

</details>

<details>

<summary>A log recorder</summary>

image

</details>

<details>

<summary>Log recorders</summary>

image

</details>

<details>

<summary>Log levels</summary>

image

</details>

### Manual testing notes

When using the core-pr-tester (`docker run --rm -ti -p 8080:8080 -e ID=4530 jenkins/core-pr-tester`), you can use script console to enable the new permission: 

`Jenkins.SYSTEM_READ.enabled = true`

### Proposed changelog entries

* Entry 1: JENKINS-61203, Allows users with system read permission to view the system logs
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by res0nance — APPROVED** [review]

**Review by jvz — APPROVED** [review]
Code looks good. No manual testing performed here.

Any documentation to update here warning administrators that exposing fine and below log recorders may compromise sensitive information depending on what gets logged. Major data leak potential there.

**Review by oleg-nenashev — COMMENTED** [review]
`doRss()` probably needs an explicit permission check as well.

**Comment by timja** [comment]
> Code looks good. No manual testing performed here.
> 
> Any documentation to update here warning administrators that exposing fine and below log recorders may compromise sensitive information depending on what gets logged. Major data leak potential there.

@jvz how about:
image

**Comment by jvz** [comment]
Looks good to me!

**Review by jeffret-b — APPROVED** [review]
Looks good. I reviewed the code and did some simple manual testing. Everything behaved as expected.

**Review by EstherAF — APPROVED** [review]
Code looks good to me. I didn't manually test it.

**Comment by daniel-beck** [comment]
CI failure: `core/src/main/java/jenkins/management/SystemLogLink.java:30:1: Illegal import - javax.annotation.Nonnull. [IllegalImport]`

**Comment by timja** [comment]
> CI failure: `core/src/main/java/jenkins/management/SystemLogLink.java:30:1: Illegal import - javax.annotation.Nonnull. [IllegalImport]`

merged in master

File: core/src/main/java/hudson/logging/LogRecorder.java [hunk 1]
```diff
@@ -401,6 +407,8 @@ public synchronized void doDoDelete(StaplerResponse rsp) throws IOException, Ser
      * RSS feed for log entries.
      */
     public void doRss( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
+        Jenkins.get().checkPermission(Jenkins.SYSTEM_READ);
```
Comment (daniel-beck) [inline core/src/main/java/hudson/logging/LogRecorder.java:410]: Redundant given https://github.com/jenkinsci/jenkins/pull/4530/files#diff-cb8b7584e7a539c988ae99e9951f7b65R213 and https://github.com/jenkinsci/jenkins/pull/4530/files?w=1#diff-fcc5c10b76c1e25e4ac9caae4fa96acfR71 ?

**Review by daniel-beck — COMMENTED** [review]
Does not protect `#doAutoCompleteLoggerName` but that seems OK.

Overall looks reasonable. I was unable to find any security related problems except the well known one about verbose logs containing secrets.

**Comment by timja** [comment]
For updating logging redirects from wiki to jenkins.io: https://github.com/jenkins-infra/jenkins.io/pull/3117

File: core/src/main/resources/hudson/logging/LogRecorder/configure.jelly [hunk 2]
```diff
@@ -27,9 +27,16 @@ THE SOFTWARE.
 -->
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
-<l:layout>
+<l:layout permission="${app.ADMINISTER}" title="${%Configure Log Recorder}">
   <st:include page="sidepanel.jelly" />
   <l:main-panel>
+    <h1>${%Configure log recorder} <t:help iconSize="large" href="https://jenkins.io/redirect/log-recorders" /></h1>
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/logging/LogRecorder/configure.jelly:33]: Same redirect as https://github.com/jenkinsci/jenkins/blob/ed80d24429a112f21fefaa09e39953c625628646/core/src/main/resources/hudson/logging/LogRecorderManager/index.jelly#L35 seems to make sense.

**Review by daniel-beck — APPROVED** [review]
👍 

(Assuming my super long warning message isn't too long now 😅 )

**Comment by timja** [comment]
> 👍
> 
> (Assuming my super long warning message isn't too long now 😅 )

Seems fine to me:

image

Let's merge this in 24 hours if there's no negative feedback

**Pull request merged**